### ISSUE TITLE
Suggestion: Provide alternate phrasing dictionary

### ISSUE THREAD

**Issue by lgraubner** [nodejs/package-maintenance#67]
A lot of common used phrasing is offending and should be avoided. For example master/slave can be replaced with primary/secondary (or anything else). See this [medium article](https://medium.com/@felixgondwe/replace-master-slave-terminology-dd7c67327669) for example. There are a lot more phrases.

I'd suggest providing a dictionary with alternate phrasing which can be discussed here or somewhere else. Package owners should be advised how to avoid pitfalls and be as inclusive as possible.

(just saw @bnb comment at #59 which leads me to suggest this)

**Comment by balupton** [comment]
"Should be avoided" is a statement of values rather than fact, which is evidenced by the nature of the problem. As such, this should be opt in per community/person, rather than enforced as a standard.

I think I saw some repos already on github that offer such listings and suggestions, some even scan your text and provide politically correct alternatives. Listing them would definitely be useful for those who share the sentiment.

**Comment by wentout** [comment]
I'm very sorry, I just can't understand.
I'm not so good in English, I think I'd better understand raw terminology than a way to reveal the meaning of what happens inside. If some prorgamm, let me say quoted, directly from my level of language "is controlled by other programm", which terminology I must use?
I didn't got this from that article.

I just trying to say it might be really hard for non natives to understand the difference between "main" and "primary". And, saying "everybody must use rulled dictionary", emm... it is hard for non native. Sorry.

**Comment by wentout** [comment]
Thank you, I understand.

And using this dictionary, should I ask somewone about some keywords usage that might be not so good as other keywords? Or this was about asking the additional explanation to some not so obvious terms?

As for me, I'm trying to get rid of that sort of terms in my native language too. However, it is really tough effort, for some terms we don't have direct translations. And, therefore, completely no linking to the context of the meanings. So we just can't translate as fast backwards, and unable to see odds which are obvious for the others. And my second question was about how can I protect others from myself if I will unfortunately fail to some abnormal terms trying to explain some hard but good.

**Comment by ljharb** [comment]
Note that node (and a growing number of projects in the industry) has a code of conduct - following it is not “opt in” or optional.

**Comment by wentout** [comment]
> Note that node (and a growing number of projects in the industry) has a code of conduct - following it is not “opt in” or optional.

Oh... then there definetely must be a list of preferable phrases...

**Comment by ljharb** [comment]
Changing language absolutely can make a package more maintainable, because it excludes fewer (or more, depending) people. Separately, contributing a line of code to a project is *in no way* a requirement to be helpful nor to be able to ask something of the maintainers - I’ve never contributed a line of code to node, but i don’t think it’s a stretch to say I’ve contributed to node a decent amount.

I definitely agree it’s out of scope of this initiative to file issues in other projects asking them to do work - but it’s very much in scope to provide guidance in best practices, which objectively include having governance, and a CoC that requires inclusive language.

**Comment by mcollina** [comment]
> Separately, contributing a line of code to a project is in no way a requirement to be helpful nor to be able to ask something of the maintainers - I’ve never contributed a line of code to node, but i don’t think it’s a stretch to say I’ve contributed to node a decent amount.

I totally agree with this. I meant individuals that are not part of the project community. I’m sorry for the confusion.

**Comment by benjamingr** [comment]
@balupton first of all, thanks for engaging and expressing your views. I'm going to take off my "package maintainer" and "core collaborator" hat and put on my moderator one for this comment.

At the moment - please consider that you have not engaged a lot with the Node.js core project before and the first time I'm seeing you in this tracker it's talking about politics. That is not to say you're not a competent engineer - only that this is my perspective. 

The question of "who's right" isn't interesting to me to be honest. It's about "what would be the best path forward for the JavaScript ecosystem" and "what would make projects more maintainable". I absolutely agree with @mcollina that it's not Node.js's place as a project to enforce its CoC across other projects or the ecosystem.

Node.js does have a CoC, we intend to enforce it and it's been working out quite well for us. The project is friendlier and more productive than it was before the CoC and we're quite happy with it as a whole. We intend to enforce the CoC across the board and moderate any comment that violates it.

That said - I ask that you reconsider your last response. You seem to feel strongly about this view. I request that you step back and consider what you actually want to accomplish here and how the way you've gone about it so far has worked out for you.

Feel free to reach out personally (my email is listed in https://github.com/nodejs/node ) if you want to discuss this further in private.

**Comment by Qard** [comment]
If you're interested in providing docs/resources around more inclusive writing styles, then I would assume PRs are welcome. 🤷‍♂️ 

Not everyone in the group is going to want to commit their own time to those efforts, and that's fair. We're all volunteering our own free time here. If you want to see something specific come from this group, you're welcome to step up and help make that happen.

**Comment by mhdawson** [comment]
@lgraubner to echo the discussion, I think this can fit in with the baseline practices being discussed in https://github.com/nodejs/package-maintenance/issues/119.

If you wanted to write up the text for this best practice along with links to resources that can be used that is probably the next concrete step (I said link to resources as it was mentioned that there are already some resources and even tools so not sure we need to develop a new list, although that still may make sense).

**Comment by rexagod** [comment]
I'd be happy to work on a parser package that points out words based on their [effective valence](https://en.wikipedia.org/wiki/Valence_(psychology)) in a file, bringing such words to the maintainer's notice. After that, I guess it's entirely up to them to modify such words or leave them as they are (hence not enforcing such modifications in any way).

**Comment by Eomm** [comment]
@rexagod Could you elaborate?

I though a document would be enough to broad awareness.
I think it is not a matter of "banned-words" only, but of inclusivity

**Comment by rexagod** [comment]
Correct me if I'm wrong but to me, this seems to be an issue regarding lessening the usage of contextual offensive (negative valence) words, by some method, be it a parser package that analyzes the code or simply just a document that outlines this idea, as you said.

What I was proposing here is a package that does the same and runs on an event (say, compilation, build, hooks, etc.) that shows the user potentially offensive words (the valence measure I proposed [should take care of inclusivity](https://github.com/words/afinn-165/blob/07c04f629345c9901d8a088aa834cf6b29f6edc6/index.json#L2766)) which they may go ahead and make changes to so as to ensure our motive here, while not forcibly levying them to do so.

The reason behind why I propose such a package is the fact that even if someone is about to commit code that does not adhere to our ideas here, be it after a long break or something (after which they could very well forget about the document), this will remind them about any such occurrences. Including this package in CI could point out to the reviewers if there is still room for such changes in the code. In the case of just a document, reviewers/users may overlook/forget about adhering to it and corrections in the future may become very tedious.

We could also keep a document with a detailed text to how one could practice naming in a _more acceptable way_, including good (and bad?) conventional practices and examples, etc. (or whatever you think the doc should include).

**Comment by Eomm** [comment]
I don't want to block you to write some code, but I feel that this utility would not hit the target at the core and, reading the [comment above](https://github.com/nodejs/package-maintenance/issues/67#issuecomment-445458781), it seems that such tools exist.

When I say "target" I refer to:
- the WG scope: a package more maintainable
- "provide guidance in best practices, which objectively include having governance, and a CoC that requires inclusive language." [ref comment](https://github.com/nodejs/package-maintenance/issues/67#issuecomment-445485071)

For these reasons, I think that a document that explains to maintainers and future maintainers why to adopt a language and a CoC file, etc.. would help the project/package. And I'm not sure how a tool could do that.
The master/slave diatribe is only the peek of the iceberg, I read some report from [ethics.org](https://www.ethics.org/) and I was speechless: I don't think at all that writing this doc will be easier than writing a tool, I think it could be interesting instead!

Anyway, I don't want to impose my idea of course.
We should argue to reach the consensus

**Comment by rexagod** [comment]
Apologies for the delay, @Eomm!

I skimmed through ethics.org and realised that the scope the this issue is way too broad to be handled by a tool. I'm all for collaborating on the document which, as is obvious, won't be any easier than the tool, but definitely worth it!

I'd love to hear any ideas/sources you think would be a good starting point for the formation of this doc, meanwhile I'll dig up a few myself!
